File: src/common/py/core/messaging/message_types_catalog.py
from .message import MessageType
from ...utils import ItemsCatalog


@ItemsCatalog.define()
class MessageTypesCatalog(ItemsCatalog[type[MessageType]]):
    """
    Global catalog of all registered message types.

    This is a globally accessible list of all message types, associated with their respective message names.
    It's mainly used to create proper message objects from incoming network messages.
    """

File: src/common/web/core/messaging/MessageTypesCatalog.ts
import { ItemsCatalog } from "../../utils/ItemsCatalog";
import { type ConstructableMessage } from "./Message";

/**
 * Global catalog of all registered message types.
 *
 * This is a globally accessible list of all message types, associated with their respective message names.
 * It's mainly used to create proper message objects from incoming network messages.
 */

@ItemsCatalog.define()
export class MessageTypesCatalog extends ItemsCatalog<ConstructableMessage> {
}
